site/src/api/queries/workspaces.ts

```diff
@@ -266,6 +266,11 @@ export const startWorkspace = (
 export const cancelBuild = (workspace: Workspace, queryClient: QueryClient) => {
 	return {
 		mutationFn: () => {
+			if (workspace.latest_build.status === "pending") {
```

Comment:
Seems like we'd either want to send `expect_status=pending` OR `expect_status=running`, but here's you're not sending anything for running builds.